static/js/containers/ProfilePage.js

```diff
@@ -40,9 +40,9 @@ class ProfilePage extends React.Component {
       dispatch(startProfileEdit());
     }
     return dispatch(validateProfile(profile, requiredFields, messages)).then(() => {
-      dispatch(saveProfile(SETTINGS.username, profile)).then(() => {
-        dispatch(clearProfileEdit());
-      });
+      return dispatch(saveProfile(SETTINGS.username, profile));
+    }).then(() => {
+      return dispatch(clearProfileEdit());
```

Comment:
what's the reason for changing these two lines? is there a reason we can't do this?

```js
return dispatch(validateProfile(profile, requiredFields, messages)).then(() => {
  return dispatch(saveProfile(SETTINGS.username, profile)).then(() => {
    dispatch(clearProfileEdit());
  });
});
```

still trying to wrap my head around promises a little more, and understand how this change relates to the change over in #314.